terraform/modules/application/events.tf

```diff
@@ -0,0 +1,43 @@
+resource "aws_cloudwatch_event_rule" "app_task_exited" {
+  name = "${var.prefix}-app-task-exited"
+
+  event_pattern = jsonencode(
+    {
+      "source" : [
+        "aws.ecs"
+      ],
+      "detail-type" : [
+        "ECS Task State Change"
+      ],
+      "detail" : {
+        "group" : [
+          "service:${aws_ecs_service.app.name}"
+        ],
+        "stoppedReason" : [
+          "Essential container in task exited"
+        ]
+      }
+    }
+  )
+}
+
+resource "aws_cloudwatch_event_rule" "app_service_action_problem" {
```

Comment:
Could comment in the stackoverflow post you mentioned to me here, if you think it would be useful for a future viewer of the code